packages/puppeteer-parse/index.js

```diff
@@ -397,15 +397,33 @@ function validateUrlString(url) {
   }
 }
 
+function tryParseUrl(urlStr) {
+  if (!urlStr) {
+    return null;
+  }
+  
+  // a regular expression to match URLs
+  var regex = /(https?:\/\/[^\s]+)/g;
+  
+  var match = regex.exec(urlStr);
+  
+  if (match) {
+    return match[1];
```

Comment:
We probably should use `String.prototype.match()` method since we want to fetch all the matches at one go

Reply:
Thanks for comment! yes, no need to get details of each match with loop, I updated it to use `match` method, and only return first match, what do you think?

Reply:
No problem at all. I think this good to 🚢